src/TraceEvent/Parsers/ClrTraceEventParser.cs

```diff
@@ -2590,16 +2629,107 @@ public override object PayloadValue(int index)
             }
         }
 
-        private event Action<GenAwareBeginTraceData> m_target;
+        private event Action<GenAwareTemplateTraceData> m_target;
         #endregion
     }
-    public sealed class GenAwareEndTraceData : TraceEvent
+
+    public sealed class GCLOHCompactInfo
     {
-        public int Count { get { return GetInt32At(0); } }
-        public int ClrInstanceID { get { return GetInt16At(4); } }
+        /// <summary>
+        /// The time spent on planing the LOH for compaction
+        /// </summary>
+        public int TimePlan
+        {
+            get
+            {
+                return m_timePlan;
+
+            }
+        }
+
+        /// <summary>
+        /// The time spent on copying objects during LOH compaction
+        /// </summary>
+        public int TimeCompact
+        {
+            get
+            {
+                return m_timeCompact;
+
+            }
+        }
+
+        /// <summary>
+        /// The time spent on relocating pointers during LOH compaction
+        /// </summary>
+        public int TimeRelocate
+        {
+            get
+            {
+                return m_timeRelocate;
+            }
+        }
+
+        /// <summary>
+        /// The total number of pointers found in the LOH during LOH compaction
+        /// </summary>
+        public long TotalRefs
+        {
+            get
+            {
+                return m_totalRefs;
+            }
+        }
+
+        /// <summary>
+        /// The total number of pointers found pointing to null in the LOH during LOH compaction
+        /// </summary>
+        public long ZeroRefs
+        {
+            get
+            {
+                return m_zeroRefs;
+            }
+        }
+
+        #region private
+        internal GCLOHCompactInfo(int timePlan, int timeCompact, int timeRelocate, long totalRefs, long zeroRefs)
+        {
+            m_timePlan = timePlan;
+            m_timeCompact = timeCompact;
+            m_timeRelocate = timeRelocate;
+            m_totalRefs = totalRefs;
+            m_zeroRefs = zeroRefs;
+        }
+
+        private int m_timePlan;
+        private int m_timeCompact;
+        private int m_timeRelocate;
+        private long m_totalRefs;
+        private long m_zeroRefs;
+        #endregion
+    }
+
+    public class GCLOHCompactTraceData : TraceEvent
```

Comment:
Can you please add a new test trace that includes these new events so that they are covered by tests?

Reply:
@brianrob, I needed some help with this. I have a program that can generate all these three types of events, but I am not sure how to add them to the tests.

The program is simple enough, it is just these few lines:

```c#
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

using System;
using System.Reflection;
using System.Runtime;
using System.Diagnostics;

namespace CoreLab
{
    internal static class Program
    {
        private static void Main()
        {
            Console.ReadLine(); // So that I have them to attach dotnet-trace
            // Here is how I can trigger the FitBucket event, the event is enable for verbose level only.
            GC.Collect(
                /* generation = */ 1,
                /* mode       = */ GCCollectionMode.Forced,
                /* blocking   = */true);

            // Here is how I can trigger the GCLOHCompact event
            GCSettings.LargeObjectHeapCompactionMode = GCLargeObjectHeapCompactionMode.CompactOnce;
            GC.Collect();
        }
    }
}
```

The `dotnet-trace` command-line needs to be tailored to contain verbose trace to get it. Here is the command line:
```sh
dotnet-trace collect --clrevents gc --clreventlevel verbose -p <pid>
```

Because of [this](https://github.com/dotnet/runtime/pull/79671) bug, the new `GCSettingRundown` event can be present only in the latest runtime after that PR is merged. For ETW traces, it should work on .net 6+, but for the trace collection side, it requires the new GC keyword I introduced in the `ClrRundownTraceEventParser.Keywords.Default` in this PR to work because otherwise, the keyword is still missing.